upload/src/addons/SV/WarningImprovements/XF/Service/Conversation/Notifier.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace SV\WarningImprovements\XF\Service\Conversation;
+
+class Notifier extends XFCP_Notifier
+{
+    protected function _canUserReceiveNotification(\XF\Entity\User $user, \XF\Entity\User $sender = null)
+    {
+        if (\SV\WarningImprovements\Listener::$warnngObj)
+        {
+            if (\SV\WarningImprovements\Listener::$warnngObj->user_id == $user->user_id)
+            {
+                $options = $this->app->options();
+
+                if ($options->sv_force_conversation_email_on_warning) // Force conversation email on warning
+                {
+                    $receiveAdminEmail = $user->Option->receive_admin_email;
+
+                    if (!$options->sv_respect_receive_admin_email_on_warning)
+                    {
+                        $receiveAdminEmail = true;
+                    }
+
+                    if ($options->sv_only_force_warning_email_on_banned && $user->is_banned)
+                    {
+                        return $receiveAdminEmail;
```

Comment:
The text for this option is 'Only send warning email if user is banned'

So shouldn't this be `if ($options->sv_only_force_warning_email_on_banned && !$user->is_banned) { $receiveAdminEmail = false; }`

Otherwise the option does nothing. As this simplifies to `return $receiveAdminEmail;` for when the `if` statement us true or false